lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1860,6 +1967,29 @@ LogicalResult FIRRTLLowering::visitExpr(SpecialConstantOp op) {
                      getOrCreateIntConstant(APInt(/*bitWidth*/ 1, op.value())));
 }
 
+LogicalResult FIRRTLLowering::visitExpr(SubindexOp op) {
+  if (isZeroBitFIRRTLType(op->getResult(0).getType()))
+    return setLowering(op, Value());
+
+  auto resultType = lowerType(op->getResult(0).getType());
+  Value value = getPossiblyInoutLoweredValue(op.input());
+  if (!resultType || !value) {
+    op.emitError() << "input lowering failed";
+    return failure();
+  }
+
+  auto iIdx = getOrCreateIntConstant(
+      getBitWidthFromVectorSize(
+          op.input().getType().cast<FVectorType>().getNumElements()),
+      op.index());
+
+  // If the value has an inout type, we need to lower to ArrayIndexInOutOp.
+  if (value.getType().isa<sv::InOutType>())
+    return setLoweringTo<sv::ArrayIndexInOutOp>(op, value, iIdx);
+  // Otherwise, hw::ArrayGetOp
+  return setLoweringTo<hw::ArrayGetOp>(op, value, iIdx);
```

Comment:
I have pushed some changes but this part is only essential difference. 
I wrongly used `hw.array_get` for all subindex ops and it was not possible to lower them as destination of connect. 

```
circuit Foo:
  module Foo:
    input clock: Clock
    input a: UInt<1>[2]
    output b: UInt<1>[2]

    b[0] <= a[0]
    b[1] <= a[1]
```

Reply:
This looks great. It also works for the case of reading from an output port. 👍